### PR TITLE
[CBRD-20787] Common table expressions

### PR DESCRIPTION
**Pull request by andrei14vl**
http://jira.cubrid.org/browse/CBRD-20787

Implementation of the common table expressions. 

I discovered an issue with print columns and I will try to fix it before merge if posssible.
Later edit: The issue was that including a simple query in a CTE affected the printed columns names. I discovered this issue for functions, where the column name is computed in pt_print_function(). The current behavior looks correct since wrapping the query in a SELECT instead of a CTE also changes the column names.

### CODE REVIEW

**Comment by cubridci** [comment]
Build comment file: 
--------------
#### File: src/query/query_executor.c (1 incorrect indentation(s) found)
```
 15463 		  save_recursive_list_id = recursive_part->list_id;
```
--------------
**Please run 'indent -l120 -lc120' to correct the indentation**

--------------

File: src/parser/xasl_generation.c [hunk 1]
```diff
@@ -16328,7 +16529,7 @@ parser_generate_xasl_proc (PARSER_CONTEXT * parser, PT_NODE * node, PT_NODE * qu
   bool query_Plan_dump_fp_open = false;
 
   /* we should propagate abort error from the server */
-  if (!parser->abort && PT_IS_QUERY (node))
+  if ((!parser->abort && PT_IS_QUERY (node)) || node->node_type == PT_CTE)
```
Comment (eseokoh) [inline src/parser/xasl_generation.c]: please confirm whether it should be
```C
  if (!parser->abort && (PT_IS_QUERY (node) || node->node_type == PT_CTE))
```

**Comment by eseokoh** [comment]
@andrei14vl 

I've added small fixes as 72b0666bb6e74dab9b529844ac1a7603df243253.
Please confirm it. :)

**Comment by eseokoh** [comment]
@andrei14vl 
2f9236aceb656d67819bcf3f900f4fa13fa3c1d0 looks correct. 
Thank you. :)

File: src/parser/xasl_generation.c (hunk 1 above)
Comment (andrei14vl) [inline src/parser/xasl_generation.c]: I changed. Thank you for pointing.

**Comment by eseokoh** [comment]
I'd like to propose to merge the patch and let QA to follow it. 
Thank you. :)

**Pull request merged**